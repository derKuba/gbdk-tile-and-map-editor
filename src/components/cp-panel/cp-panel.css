:host {
  display: flex;
  justify-content: space-between;
  width: 160px;
  margin-left: 50px;
  margin-top: 17px;
}

.box {
  border: 1px solid black;
  border-radius: 5px;
  background-color: grey;
  padding: 5px  5px 1px 5px;
  margin-left: 11px;
}

.cp-icon {
  width: 20px;
  height: 20px;
}


.box:last-of-type {
  margin: 0 26px 0 0;
}