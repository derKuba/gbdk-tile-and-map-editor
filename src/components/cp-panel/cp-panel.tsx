import { Component, Host, h, Prop } from '@stencil/core';

import Tunnel from '../../provider/StateProvider';

enum ACTIONS {
    COPY = 0,
    PASTE = 1,
}

@Component({
    tag: 'mvjp-cp-panel',
    styleUrl: 'cp-panel.css',
    shadow: true,
})
export class CpPanel {
    /**
     *  Set ID of actual matrix to copy
     */
    @Prop() setCopyIndex: Function;

    /**
     *  Paste Action
     */
    @Prop() pasteMatrix: Function;

    private copyOrPaste = action => {
        if (action === ACTIONS.COPY) {
            this.setCopyIndex();
        }
        if (action === ACTIONS.PASTE) {
            this.pasteMatrix();
        }
    };

    render() {
        return (
            <Host>
                <div class="box icon-alignment">
                    <img
                        class="cp-icon"
                        onClick={() => this.copyOrPaste(ACTIONS.COPY)}
                        src={'./assets/icon/freepik_document.png'}
                        alt="arrow turn around icon"
                        title="Copy Matrix"
                    />
                </div>

                <div class="box icon-alignment">
                    <img
                        class="cp-icon"
                        onClick={() => this.copyOrPaste(ACTIONS.PASTE)}
                        src={'./assets/icon/freepik_clipboard.png'}
                        alt="arrow trash icon"
                        title="Paste"
                    />
                </div>
            </Host>
        );
    }
}

Tunnel.injectProps(CpPanel, ['setCopyIndex', 'pasteMatrix']); // eslint-disable-line @stencil/ban-side-effects
