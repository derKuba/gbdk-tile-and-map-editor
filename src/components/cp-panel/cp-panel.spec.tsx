import { newSpecPage } from '@stencil/core/dist/testing';
import { h } from '@stencil/core';

import { CpPanel } from './cp-panel';

describe('mvjp-cp-panel', () => {
    it('builds', () => {
        expect(new CpPanel()).toBeTruthy();
    });

    it('matches snapshot', async () => {
        const page = await newSpecPage({
            components: [CpPanel],
            template: () => {
                return <mvjp-cp-panel />;
            },
        });

        expect(page.root).toMatchSnapshot();
    });

    it('should test clickhandler', async () => {
        const pasteMatrix = jest.fn();
        const setCopyIndex = jest.fn();

        const page = await newSpecPage({
            components: [CpPanel],
            template: () => {
                return <mvjp-cp-panel setCopyIndex={setCopyIndex} pasteMatrix={pasteMatrix}></mvjp-cp-panel>;
            },
        });

        page.root.shadowRoot.querySelector('img[title="Copy Matrix"]').click();
        await page.waitForChanges();
        expect(setCopyIndex).toBeCalled();

        page.root.shadowRoot.querySelector('img[title="Paste"]').click();
        await page.waitForChanges();
        expect(pasteMatrix).toBeCalled();
    });
});
