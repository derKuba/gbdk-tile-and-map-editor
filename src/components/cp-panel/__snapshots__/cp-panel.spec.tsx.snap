// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mvjp-cp-panel matches snapshot 1`] = `
<mvjp-cp-panel>
  <mock:shadow-root>
    <div class="box icon-alignment">
      <img alt="arrow turn around icon" class="cp-icon" src="./assets/icon/freepik_document.png" title="Copy Matrix">
    </div>
    <div class="box icon-alignment">
      <img alt="arrow trash icon" class="cp-icon" src="./assets/icon/freepik_clipboard.png" title="Paste">
    </div>
  </mock:shadow-root>
</mvjp-cp-panel>
`;
